app/javascript/components/mentoring/Queue/SolutionList.jsx

```diff
@@ -2,6 +2,7 @@ import React from 'react'
 import { Solution } from './Solution'
 import { Pagination } from '../Pagination'
 import { usePaginatedRequestQuery } from '../../../hooks/request-query'
+import { Loading } from '../../common/Loading'
 
 export function SolutionList({ request, setPage }) {
   const { status, resolvedData, latestData } = usePaginatedRequestQuery(
```

Comment:
Maybe also use the `isSuccess, isError, isFetching` fields here instead of status?